### DISCUSSION TITLE
Surface roughness in 2D

### DISCUSSION THREAD

**Discussion by mauroarcidiacono** [neperfepx/neper#618] [Q&A] [answered]
Hi,

Would it be possible to use Neper to establish a geometry and then define the grains and the mesh inside of it? For example, would it be possible to create grains inside of the following 2D figure?

![roughness](https://user-images.githubusercontent.com/76496290/204545301-a20310a9-4830-4522-b300-b8ecc7623d5b.jpg)
 
I would like to include the effects of surface roughness in a simulation. 

Thank you very much for your help in advance.

**Comment by rquey** [comment, marked as answer]
It is possible to cut a tessellation using `-transform cut`. So, it would work the other way around: generate the tessellation in a usual (convex) domain and then cut it.  In principle, this may work, but it will be difficult to generate an irregular boundary like on your image.

A question is how the characteristic lengths of the local fluctuations compare to the grain size. It would have to be significantly larger for this option to work.

If the local fluctuations are small compared to the grain size, they you could simply work on the mesh, by shifting the nodes...

**Reply by mauroarcidiacono** [reply]
Thank you very much for your recommendations Romain. I ended up creating a script that uses the .tess file from Neper to create a part in Abaqus without mesh with the corresponding material properties. Then, I cut the part with the desired roughness profile and I meshed it.
